lib/_http_client.js

```diff
@@ -360,9 +360,10 @@ function socketCloseListener() {
   // NOTE: It's important to get parser here, because it could be freed by
   // the `socketOnData`.
   var parser = socket.parser;
-  req.emit('close');
+
   if (req.res && req.res.readable) {
     // Socket closed before we emitted 'end' below.
+    req.emit('aborted');
```

Comment:
Is this correct? This branch means there's a response coming but we haven't received its end. I don't think this qualifies for an 'aborted' event for the request (only the response).

If this is motivated by how this works in h2, it's slightly different there because you have the request & response in one (duplex stream), instead of separate like with http1.

Reply:
Well, from an api user's perspective the response could conceptually be part of the request (think http2)?

Reply:
I think both behaviours make equal sense. I would prefer the h2 way just to avoid confusion.